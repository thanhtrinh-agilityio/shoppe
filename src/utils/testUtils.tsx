import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
