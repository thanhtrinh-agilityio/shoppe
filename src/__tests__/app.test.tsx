import React from 'react'
import { NextPage } from 'next'
import { Router } from 'next/router'

// Pages
import App from '@pages/_app'

// Utils
import { render } from '@utils/testUtils'

describe('App render', () => {
  const mockedPageProps = {}
  const mockComponent: NextPage = () => <div>Mocked Test Component</div>
  const mockRouterFunction = jest.fn()

  it('Should show match App DOM Snapshot', () => {
    const component = render(
      <App
        pageProps={mockedPageProps}
        Component={mockComponent}
        router={mockRouterFunction as unknown as Router}
      />,
    )

    expect(component).toMatchSnapshot()
  })
})
