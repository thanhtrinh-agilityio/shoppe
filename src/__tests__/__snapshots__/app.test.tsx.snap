// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render Should show match App DOM Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <header
        class="css-n5jh5q"
      >
        <h1
          class="chakra-heading css-1dklj6k"
        >
          <img
            alt="Logo Shoppe"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/images/logo.png?w=256&q=75"
            srcset="/images/logo.png?w=128&q=75 1x, /images/logo.png?w=256&q=75 2x"
            style="color: transparent;"
            width="125"
          />
        </h1>
        <div
          class="css-79elbk"
        >
          <div
            class="css-c3i4bm"
          >
            <img
              alt="Shopping Cart Icon"
              data-nimg="fill"
              data-testid="image"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/images/shoppingCart.svg?w=3840&q=75"
              srcset="/images/shoppingCart.svg?w=640&q=75 640w, /images/shoppingCart.svg?w=750&q=75 750w, /images/shoppingCart.svg?w=828&q=75 828w, /images/shoppingCart.svg?w=1080&q=75 1080w, /images/shoppingCart.svg?w=1200&q=75 1200w, /images/shoppingCart.svg?w=1920&q=75 1920w, /images/shoppingCart.svg?w=2048&q=75 2048w, /images/shoppingCart.svg?w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="css-2jnr0l"
          >
            <p
              class="chakra-text css-1xb8dar"
            >
              0
            </p>
          </div>
        </div>
      </header>
      <main
        class="css-qcp6ks"
      >
        <div>
          Mocked Test Component
        </div>
      </main>
      <footer
        class="css-xmbrgk"
      >
        <div
          class="css-gg4vpm"
        >
          <nav
            class="css-fd0xex"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
            <a
              href="/terms-of-services"
            >
              Terms of services
            </a>
            <a
              href="/shipping-and-return"
            >
              Shipping and Return
            </a>
          </nav>
          <div
            class="css-11xbhwq"
          >
            <p
              class="chakra-text css-ozfm6u"
            >
              Give an email, get the newsletter.
            </p>
            <div
              class="css-1o1su5v"
            >
              <img
                alt="Right arrow"
                data-nimg="fill"
                data-testid="image"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/images/iconRight.svg?w=3840&q=75"
                srcset="/images/iconRight.svg?w=640&q=75 640w, /images/iconRight.svg?w=750&q=75 750w, /images/iconRight.svg?w=828&q=75 828w, /images/iconRight.svg?w=1080&q=75 1080w, /images/iconRight.svg?w=1200&q=75 1200w, /images/iconRight.svg?w=1920&q=75 1920w, /images/iconRight.svg?w=2048&q=75 2048w, /images/iconRight.svg?w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
          </div>
        </div>
        <div
          class="css-167212l"
        >
          <p
            class="chakra-text css-tujbob"
          >
            <b
              class="chakra-text css-lz4kvn"
            >
              Â© 2021 Shelly.
            </b>
             
            Terms of use
            <b
              class="chakra-text css-lz4kvn"
            >
               
              and
            </b>
             
            privacy policy
          </p>
          <div
            class="css-st3z69"
          >
            <div
              class="css-14b8luf"
            >
              <img
                alt="LinkIn Icon"
                data-nimg="fill"
                data-testid="image"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/images/inIcon.svg?w=3840&q=75"
                srcset="/images/inIcon.svg?w=640&q=75 640w, /images/inIcon.svg?w=750&q=75 750w, /images/inIcon.svg?w=828&q=75 828w, /images/inIcon.svg?w=1080&q=75 1080w, /images/inIcon.svg?w=1200&q=75 1200w, /images/inIcon.svg?w=1920&q=75 1920w, /images/inIcon.svg?w=2048&q=75 2048w, /images/inIcon.svg?w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="css-14b8luf"
            >
              <img
                alt="Facebook icon"
                data-nimg="fill"
                data-testid="image"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/images/facebook.svg?w=3840&q=75"
                srcset="/images/facebook.svg?w=640&q=75 640w, /images/facebook.svg?w=750&q=75 750w, /images/facebook.svg?w=828&q=75 828w, /images/facebook.svg?w=1080&q=75 1080w, /images/facebook.svg?w=1200&q=75 1200w, /images/facebook.svg?w=1920&q=75 1920w, /images/facebook.svg?w=2048&q=75 2048w, /images/facebook.svg?w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="css-14b8luf"
            >
              <img
                alt="Instagram icon"
                data-nimg="fill"
                data-testid="image"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/images/instagram.svg?w=3840&q=75"
                srcset="/images/instagram.svg?w=640&q=75 640w, /images/instagram.svg?w=750&q=75 750w, /images/instagram.svg?w=828&q=75 828w, /images/instagram.svg?w=1080&q=75 1080w, /images/instagram.svg?w=1200&q=75 1200w, /images/instagram.svg?w=1920&q=75 1920w, /images/instagram.svg?w=2048&q=75 2048w, /images/instagram.svg?w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="css-14b8luf"
            >
              <img
                alt="Twitter icon"
                data-nimg="fill"
                data-testid="image"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/images/twitter.svg?w=3840&q=75"
                srcset="/images/twitter.svg?w=640&q=75 640w, /images/twitter.svg?w=750&q=75 750w, /images/twitter.svg?w=828&q=75 828w, /images/twitter.svg?w=1080&q=75 1080w, /images/twitter.svg?w=1200&q=75 1200w, /images/twitter.svg?w=1920&q=75 1920w, /images/twitter.svg?w=2048&q=75 2048w, /images/twitter.svg?w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
          </div>
        </div>
      </footer>
      <span
        hidden=""
        id="__chakra_env"
      />
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="css-n5jh5q"
    >
      <h1
        class="chakra-heading css-1dklj6k"
      >
        <img
          alt="Logo Shoppe"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/images/logo.png?w=256&q=75"
          srcset="/images/logo.png?w=128&q=75 1x, /images/logo.png?w=256&q=75 2x"
          style="color: transparent;"
          width="125"
        />
      </h1>
      <div
        class="css-79elbk"
      >
        <div
          class="css-c3i4bm"
        >
          <img
            alt="Shopping Cart Icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/images/shoppingCart.svg?w=3840&q=75"
            srcset="/images/shoppingCart.svg?w=640&q=75 640w, /images/shoppingCart.svg?w=750&q=75 750w, /images/shoppingCart.svg?w=828&q=75 828w, /images/shoppingCart.svg?w=1080&q=75 1080w, /images/shoppingCart.svg?w=1200&q=75 1200w, /images/shoppingCart.svg?w=1920&q=75 1920w, /images/shoppingCart.svg?w=2048&q=75 2048w, /images/shoppingCart.svg?w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="css-2jnr0l"
        >
          <p
            class="chakra-text css-1xb8dar"
          >
            0
          </p>
        </div>
      </div>
    </header>
    <main
      class="css-qcp6ks"
    >
      <div>
        Mocked Test Component
      </div>
    </main>
    <footer
      class="css-xmbrgk"
    >
      <div
        class="css-gg4vpm"
      >
        <nav
          class="css-fd0xex"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
          <a
            href="/terms-of-services"
          >
            Terms of services
          </a>
          <a
            href="/shipping-and-return"
          >
            Shipping and Return
          </a>
        </nav>
        <div
          class="css-11xbhwq"
        >
          <p
            class="chakra-text css-ozfm6u"
          >
            Give an email, get the newsletter.
          </p>
          <div
            class="css-1o1su5v"
          >
            <img
              alt="Right arrow"
              data-nimg="fill"
              data-testid="image"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/images/iconRight.svg?w=3840&q=75"
              srcset="/images/iconRight.svg?w=640&q=75 640w, /images/iconRight.svg?w=750&q=75 750w, /images/iconRight.svg?w=828&q=75 828w, /images/iconRight.svg?w=1080&q=75 1080w, /images/iconRight.svg?w=1200&q=75 1200w, /images/iconRight.svg?w=1920&q=75 1920w, /images/iconRight.svg?w=2048&q=75 2048w, /images/iconRight.svg?w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </div>
      </div>
      <div
        class="css-167212l"
      >
        <p
          class="chakra-text css-tujbob"
        >
          <b
            class="chakra-text css-lz4kvn"
          >
            Â© 2021 Shelly.
          </b>
           
          Terms of use
          <b
            class="chakra-text css-lz4kvn"
          >
             
            and
          </b>
           
          privacy policy
        </p>
        <div
          class="css-st3z69"
        >
          <div
            class="css-14b8luf"
          >
            <img
              alt="LinkIn Icon"
              data-nimg="fill"
              data-testid="image"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/images/inIcon.svg?w=3840&q=75"
              srcset="/images/inIcon.svg?w=640&q=75 640w, /images/inIcon.svg?w=750&q=75 750w, /images/inIcon.svg?w=828&q=75 828w, /images/inIcon.svg?w=1080&q=75 1080w, /images/inIcon.svg?w=1200&q=75 1200w, /images/inIcon.svg?w=1920&q=75 1920w, /images/inIcon.svg?w=2048&q=75 2048w, /images/inIcon.svg?w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="css-14b8luf"
          >
            <img
              alt="Facebook icon"
              data-nimg="fill"
              data-testid="image"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/images/facebook.svg?w=3840&q=75"
              srcset="/images/facebook.svg?w=640&q=75 640w, /images/facebook.svg?w=750&q=75 750w, /images/facebook.svg?w=828&q=75 828w, /images/facebook.svg?w=1080&q=75 1080w, /images/facebook.svg?w=1200&q=75 1200w, /images/facebook.svg?w=1920&q=75 1920w, /images/facebook.svg?w=2048&q=75 2048w, /images/facebook.svg?w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="css-14b8luf"
          >
            <img
              alt="Instagram icon"
              data-nimg="fill"
              data-testid="image"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/images/instagram.svg?w=3840&q=75"
              srcset="/images/instagram.svg?w=640&q=75 640w, /images/instagram.svg?w=750&q=75 750w, /images/instagram.svg?w=828&q=75 828w, /images/instagram.svg?w=1080&q=75 1080w, /images/instagram.svg?w=1200&q=75 1200w, /images/instagram.svg?w=1920&q=75 1920w, /images/instagram.svg?w=2048&q=75 2048w, /images/instagram.svg?w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="css-14b8luf"
          >
            <img
              alt="Twitter icon"
              data-nimg="fill"
              data-testid="image"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/images/twitter.svg?w=3840&q=75"
              srcset="/images/twitter.svg?w=640&q=75 640w, /images/twitter.svg?w=750&q=75 750w, /images/twitter.svg?w=828&q=75 828w, /images/twitter.svg?w=1080&q=75 1080w, /images/twitter.svg?w=1200&q=75 1200w, /images/twitter.svg?w=1920&q=75 1920w, /images/twitter.svg?w=2048&q=75 2048w, /images/twitter.svg?w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </div>
      </div>
    </footer>
    <span
      hidden=""
      id="__chakra_env"
    />
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
